### PR TITLE
Bug 45

### PR DESCRIPTION
**Pull request by asp78**
Pretty simple to fix the tests, but it required a bit of refactoring around them.

I also added to the makefile in this since I noticed the .pyc files weren't being removed in the make clean command.

Had to add a 'deleteUser' function to properly test one of the features but she ain't used anywhere in the program yet.

Ya'll need to be more careful when putting tests in to make sure they are properly doing something.

ALSO I WANT THIS TO GET IN BEFORE JON FINISHES UP HIS STORY SO HE CAN HAVE BETTER TESTS AND TOOLS TO MAKE MORE SO LETS MERGE THIS BAD BOY BEFORE HIS. IT'LL ALSO SAVE SOMEONE A MERGE SO DO IT JUST DO IT.

### CODE REVIEW

File: Backend/tests/test_user_ctrl.py [hunk 1]
```diff
@@ -88,19 +92,17 @@ def setUpClass(self):
...
-    self.assertTrue(response)'''
+    self.assertTrue(response)
 
-  def test_isEmailInSystem_fail(self):
-    print("Running isEmailInSystem fail case Test")
+  def test_isEmailNotInSystem(self):
+    print("Running isEmailNotInSystem Test")
 
-    email = "no_test@mentii.com"
+    email = "notInDB@mentii.com"
```
Comment (jonathanmcdaniel) [inline Backend/tests/test_user_ctrl.py]: Minor but we are mentii.me not mentii.com so if notInDB was a real user at mentii.com they would be getting that email. Might be worth changing.

File: Backend/tests/test_user_ctrl.py [hunk 2]
```diff
@@ -115,24 +117,35 @@ def test_addUserAndSendEmail(self):
...
+
+  # add a real test case for failing to add a user
 
   def test_register(self):
     print("Running register Test")
 
-    jsonData = {"email" : "mail@email.com"}
-    activationId = usr.register(jsonData,mail,dynamodb)
-    self.assertEqual(activationId, activationId)
+    email = "mail@email.com"
```
Comment (jonathanmcdaniel) [inline Backend/tests/test_user_ctrl.py]: We should probably only use @mentii.me emails just to be safe

File: Backend/tests/test_user_ctrl.py [hunk 3]
```diff
@@ -141,21 +154,16 @@ def test_hashPassword(self):
...
 
-  def test_activate_fail(self):
-    print("Running activate fail case Test")
+    isUserActive = 'status' in response.payload.keys() and response.payload['status'] == 'Success'
+    self.assertTrue(isUserActive)
 
-    activationId = "none"
-    response = usr.activate(activationId, dynamodb)
-    self.assertNotEqual(response, None)
+  # add a real test case for failing to add a user
```
Comment (jonathanmcdaniel) [inline Backend/tests/test_user_ctrl.py]: Is this a TODO?

File: Backend/tests/test_user_ctrl.py [hunk 4]
```diff
@@ -115,24 +117,35 @@ def test_addUserAndSendEmail(self):
...
+    # Delete user after test
+    response = usr.deleteUser(email, dynamodb)
 
-    email = "email"
-    password = "password8"
-    activationId = usr.addUserAndSendEmail(email,password,mail,dynamodb)
-    self.assertEqual(activationId,None)
+  # add test case where existing user with the same email is given
+
+  # add a real test case for failing to add a user
```
Comment (jonathanmcdaniel) [inline Backend/tests/test_user_ctrl.py]: Are these comments both TODOs? Or is there code im not seeing

File: Backend/tests/test_user_ctrl.py (hunk 1 above)
Comment (asp78) [inline Backend/tests/test_user_ctrl.py]: I'mma change em, didn't notice that lol was just re-using what was there

File: Backend/tests/test_user_ctrl.py (hunk 2 above)
Comment (asp78) [inline Backend/tests/test_user_ctrl.py]: Changing

**Pull request merged**